src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -69,6 +69,76 @@ protected override ImmutableArray<LocalSymbol> BuildLocals()
             }
         }
 
+        internal void CollectLocalsFromDeconstruction(
```

Comment:
>internal [](start = 8, length = 8)

I suspect this can be consolidated with the related code in ExpressionVariableFinder.